x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_logs/filter_dataset.tsx

```diff
@@ -6,14 +6,20 @@
  */
 
 import React, { memo, useState, useEffect, useCallback } from 'react';
-import { EuiPopover, EuiFilterButton, EuiFilterSelectItem } from '@elastic/eui';
+import styled from 'styled-components';
+import type { EuiSelectableOption } from '@elastic/eui';
+import { EuiPopover, EuiFilterButton, EuiSelectable } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import type { DataViewField, FieldSpec } from '@kbn/data-views-plugin/public';
 
 import { useStartServices } from '../../../../../hooks';
 
 import { AGENT_LOG_INDEX_PATTERN, DATASET_FIELD, AGENT_DATASET } from './constants';
 
+const StyledEuiSelectable = styled(EuiSelectable)`
+  width: 220px;
```

Comment:
I'm having trouble making the selectable options be as wide as the longest option automatically, added a workaround for a fixed width, I'm not sure if there is a better solution. Tried the different combinations of the truncate options: https://eui.elastic.co/#/forms/selectable#truncation
@breehall Hey, any suggestions? Previously with `EuiFilterSelectItem` the select expanded automatically, without truncating the options. 
The `textWrap: wrap` option doesn't seem to do anything either, and I didn't find a better solution.

Before:
<img width="233" alt="image" src="https://github.com/elastic/kibana/assets/90178898/c8db5384-0eb0-4ade-aacc-08f2661ecdaa">

After (without fixed width):
<img width="194" alt="image" src="https://github.com/elastic/kibana/assets/90178898/044019f3-dfb2-4990-b8d3-4f69042c175b">

Reply:
@breehall Another UI issue I noticed is when scrolling down the page with a long list of agents with a `EuiSelectable` component open: the list of options is visible in front of the top banner. 

<img width="1381" alt="image" src="https://github.com/elastic/kibana/assets/90178898/1e58ae7c-7d24-4421-99de-9a5f28cd5157">

Reply:
Using the `minWidth` prop works - another alternative is if you know for certain that your selectable content is fairly short, you can use custom `width: fit-content;` CSS instead. Overall, I think `minWidth` is a safer bet in case your content changes to something super long in the future that _should_ be truncated.

> when scrolling down the page with a long list of agents with a EuiSelectable component open: the list of options is visible in front of the top banner

If you know for certain your popover should sit below the sticky header/nav, you can customize this on your end with something like:

```
const { euiTheme } = useEuiTheme();

<EuiPopover zIndex={euiTheme.levels.header - 1}>
```

EUI defaults to showing popovers above headers for maximum visibility and because we can't know how the specific intention of end-popover-usages ahead of time. A couple of related issues discussing the difficulty/complexity of this topic: https://github.com/elastic/eui/issues/4756, https://github.com/elastic/eui/issues/4872

Reply:
thanks for the suggestions! I'll make the changes